Gems/SceneProcessing/Code/Source/Config/Components/SceneProcessingConfigSystemComponent.cpp

```diff
@@ -39,23 +39,56 @@ namespace AZ
-            // Defaults in case there's no config setup
-            m_softNames.push_back(aznew NodeSoftNameSetting("^.*_[Ll][Oo][Dd]1(_optimized)?$", PatternMatcher::MatchApproach::Regex, "LODMesh1", true));
-            m_softNames.push_back(aznew NodeSoftNameSetting("^.*_[Ll][Oo][Dd]2(_optimized)?$", PatternMatcher::MatchApproach::Regex, "LODMesh2", true));
-            m_softNames.push_back(aznew NodeSoftNameSetting("^.*_[Ll][Oo][Dd]3(_optimized)?$", PatternMatcher::MatchApproach::Regex, "LODMesh3", true));
-            m_softNames.push_back(aznew NodeSoftNameSetting("^.*_[Ll][Oo][Dd]4(_optimized)?$", PatternMatcher::MatchApproach::Regex, "LODMesh4", true));
-            m_softNames.push_back(aznew NodeSoftNameSetting("^.*_[Ll][Oo][Dd]5(_optimized)?$", PatternMatcher::MatchApproach::Regex, "LODMesh5", true));
-            m_softNames.push_back(aznew NodeSoftNameSetting("^.*_[Pp][Hh][Yy][Ss](_optimized)?$", PatternMatcher::MatchApproach::Regex, "PhysicsMesh", true));
-            m_softNames.push_back(aznew NodeSoftNameSetting("_ignore", PatternMatcher::MatchApproach::PostFix, "Ignore", false));
-            // If the filename ends with "_anim" this will mark all nodes as "Ignore" unless they're derived from IAnimationData. This will
-            // cause only animations to be exported from the source scene file even if there's other data available.
-            m_softNames.push_back(aznew FileSoftNameSetting("_anim", PatternMatcher::MatchApproach::PostFix, "Ignore", false,
-                { FileSoftNameSetting::GraphType(SceneAPI::DataTypes::IAnimationData::TYPEINFO_Name()) }));
 
+            PopulateDefaultSoftNameSettings();
             m_UseCustomNormals = true;
         }
 
+        void SceneProcessingConfigSystemComponent::PopulateDefaultSoftNameSettings()
+        {
+            auto settingsRegistry = AZ::SettingsRegistry::Get();
+            if (settingsRegistry)
+            {
+                AZStd::vector<NodeSoftNameSetting*> nodeSoftNameSettings;
+                settingsRegistry->GetObject(nodeSoftNameSettings, AssetProcessorDefaultNodeSoftNameSettingsKey);
+                for (NodeSoftNameSetting* nodeSoftNameSetting : nodeSoftNameSettings)
+                {
+                    if (!AddSoftName(nodeSoftNameSetting))
+                    {
+                        delete nodeSoftNameSetting;
+                        nodeSoftNameSetting = nullptr;
+                    }
+                }
+
+                AZStd::vector<FileSoftNameSetting*> fileSoftNameSettings;
+                settingsRegistry->GetObject(fileSoftNameSettings, AssetProcessorDefaultFileSoftNameSettingsKey);
+                for (FileSoftNameSetting* fileSoftNameSetting : fileSoftNameSettings)
+                {
+                    if (!AddSoftName(fileSoftNameSetting))
+                    {
+                        delete fileSoftNameSetting;
+                        fileSoftNameSetting = nullptr;
+                    }
+                }
```

Comment:
suggestion: These two blocks of code look very similar, I wonder if it'd be possible to move the logic to a small function (or lambda) you could then pass the specific arguments to as parameters?